### DISCUSSION TITLE
Best Practice DB Connection

### DISCUSSION THREAD

**Discussion by jtyme** [nextgenhealthcare/connect#5990] [Q&A] [answered]
what is the best practice for opening and closing a DB connection. Our current environment have multiple independant channels that read in flat HL7 messages flat files. We process each message seperate so each message flows through the source to the destinations. We are opening and closing DB connections multiple times a message. What is best way so we are not opening and closing so much.  I would like to open a connection when the process starts and after all messages are complete close the connection. I tried using the pre and post processing scripts but the post closes after every message.

Hope this makes sense.

**Comment by pacmano1** [comment, marked as answer]
I think many people use 1..n cached connections in $g.   Assuming you are calling DBs in Javascript here not using DB Reader and Writers.

See https://github.com/pacmano1/Mirth-Snippets/blob/main/executeSharedDBStatement.js

Another interesting way is https://github.com/MichaelLeeHobbs/connect-pgpool
